src/main.cpp

```diff
@@ -2571,19 +2570,11 @@ bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& sta
     if (pcheckpoint && nHeight < pcheckpoint->nHeight)
         return state.DoS(100, error("%s: forked chain older than last checkpoint (height %d)", __func__, nHeight));
 
-    // Reject block.nVersion=1 blocks when 95% (75% on testnet) of the network has upgraded:
```

Comment:
Well, if we have 10 voted softfork at the same time I prefer to have a loop over 10 practically identical unrolled ifs...
I would also like to have a "time proof" solution for forks that are enforced by height instead of voting as suggested in #5966 . This is the only place where you can have a generic solution for block versions.
Anyway, I don't think 
```
// Reject block.nVersion=n blocks when 95% (75% on testnet) of the network has upgraded (last version=3)
```
is less clear than 
```
// Reject block.nVersion=1 blocks when 95% (75% on testnet) of the network has upgraded:
// Reject block.nVersion=2 blocks when 95% (75% on testnet) of the network has upgraded:
```
just less repetitive. But if you have a better idea I'm happy to change the comments.
Or are you completely against the loop?